src/config/mod.rs

```diff
@@ -49,72 +56,23 @@ impl Config {
     }
 }
 
-/// A serde deserialization method to deserialize a subnet path string into a [`SubnetID`].
-fn deserialize_path<'de, D>(deserializer: D) -> Result<SubnetID, D::Error>
-where
-    D: Deserializer<'de>,
-{
-    struct SubnetIDVisitor;
-    impl<'de> serde::de::Visitor<'de> for SubnetIDVisitor {
-        type Value = SubnetID;
-
-        fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
-            formatter.write_str("a string")
-        }
-
-        fn visit_str<E>(self, v: &str) -> std::result::Result<Self::Value, E>
-        where
-            E: Error,
-        {
-            SubnetID::from_str(v).map_err(E::custom)
-        }
-    }
-    deserializer.deserialize_str(SubnetIDVisitor)
-}
-
-/// A serde deserialization method to deserialize a list of account strings into a vector of
-/// [`Address`].
-fn deserialize_accounts<'de, D>(deserializer: D) -> Result<Vec<Address>, D::Error>
-where
-    D: Deserializer<'de>,
-{
-    struct AddressSeqVisitor;
-    impl<'de> serde::de::Visitor<'de> for AddressSeqVisitor {
-        type Value = Vec<Address>;
-
-        fn expecting(&self, formatter: &mut Formatter) -> std::fmt::Result {
-            formatter.write_str("a sequence of strings")
-        }
-
-        fn visit_seq<A>(self, mut seq: A) -> std::result::Result<Self::Value, A::Error>
-        where
-            A: SeqAccess<'de>,
-        {
-            let mut vec: Vec<Address> = Vec::new();
-            while let Some(value) = seq.next_element::<String>()? {
-                let a = Address::from_str(value.as_str()).map_err(Error::custom)?;
-                vec.push(a);
-            }
-            Ok(vec)
-        }
-    }
-    deserializer.deserialize_str(AddressSeqVisitor)
-}
-
 #[cfg(test)]
 mod tests {
+    use std::net::SocketAddr;
     use std::str::FromStr;
 
     use fvm_shared::address::Address;
     use indoc::formatdoc;
-    use ipc_sdk::subnet_id::{SubnetID, ROOTNET_ID};
+    use ipc_sdk::subnet_id::{ROOTNET_ID, SubnetID};
     use url::Url;
 
     use crate::config::Config;
 
     #[test]
     fn read_config() {
```

Comment:
Instead of a single monolithic test method, I suggest have separate test methods for the different config section. For example, keep the existing method and rename it to `read_config_subnets` and add a new method called `read_config_server` to test the server-specific logic.

Reply:
@hmoniz This might have a problem as the fields must be present, if separate into two methods, both the `subnet` and `server` needs to be in the raw string, which seems to be strange.

Reply:
At least based on the current implementation, it is not possible to separate into sub tests. The other way to work around this is having:
```rust
pub struct Config {
    pub server: Option<Server>,
    pub subnets: Option<HashMap<String, Subnet>>,
}
```
Then whoever needs the specific section, just do `let server_config = config.server.expect("config not defined")`. Then we can achieve the above. 

@hmoniz @adlrocha This approach ok?

Reply:
I am not a fan of using an `Option` if we want the config to be mandatory, it may lead to future issues. Something we could do is to have a single config template for the whole test mode and all tests and then write individual tests for each field we are testing, where we can modify the tested field if needed.

In any case, no strong opinions here, let's wait for @hmoniz input.

Reply:
Let's go with @adlrocha's suggestion here. I think it strikes the right balance.